synapse/handlers/e2e_keys.py

```diff
@@ -122,6 +128,44 @@ def query_devices(self, query_body, timeout):
         @defer.inlineCallbacks
         def do_remote_query(destination):
             destination_query = remote_queries_not_in_cache[destination]
+
+            # We first consider whether we wish to update the dive list cache with
+            # the users device list. We want to track a user's devices when the
+            # authenticated user shares a room with the queried user and the query
+            # has not specified a particular device.
+            # If we update the cache for the queried user we remove them from further
+            # queries. We use the more efficient batched query_client_keys for all
+            # remaining users
+            user_ids_updated = []
+            for (user_id, device_list) in destination_query.items():
+                if user_id not in user_ids_updated:
+                    try:
+                        with PreserveLoggingContext(LoggingContext.current_context()):
+                            room_ids = yield self.store.get_rooms_for_user(user_id)
+                            if not device_list and room_ids:
+                                user_devices = yield self.device_handler.device_list_updater.user_device_resync(
```

Comment:
A comment along the lines of "we should be receiving updates for the user, but haven't done an initial sync of the device list, so we do so now"